### PR TITLE
add getFlowData method to MyAvatar

### PR DESCRIPTION
**Pull request by luiscuenca**
https://highfidelity.manuscript.com/f/cases/21097/Update-Flow-App-to-use-new-C-flow-API
This PR adds a method to access flow data to support flowApp

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17990/HighFidelity-Beta-PR15141-e8da6b5.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17990/HighFidelity-Quest-Beta-PR15141-e8da6b5.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17990/HighFidelity-Beta-PR15141-e8da6b5.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17990/HighFidelity-Beta-Interface-PR15141-e8da6b5.exe

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18004/HighFidelity-Beta-PR15141-f363d95.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18004/HighFidelity-Quest-Beta-PR15141-f363d95.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18004/HighFidelity-Beta-PR15141-f363d95.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18004/HighFidelity-Beta-Interface-PR15141-f363d95.exe

File: interface/src/avatar/MyAvatar.cpp [hunk 1]
```diff
@@ -5387,6 +5391,82 @@ void MyAvatar::useFlow(bool isActive, bool isCollidable, const QVariantMap& phys
...
+            threadData.insert(thread._jointsPointer->at(thread._joints[0]).getName(), indices);
+        }
+        flowData.insert("physics", physicsData);
+        flowData.insert("collisions", collisionsData);
+        flowData.insert("threads", threadData);
+    }
+    return flowData;
+}
+
+QVariantList MyAvatar::getCollidingFlowJoints() {
```
Comment (SamGondelman) [inline interface/src/avatar/MyAvatar.cpp]: These 3 methods don't look threadsafe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18010/HighFidelity-Beta-PR15141-4858f64.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18010/HighFidelity-Beta-Interface-PR15141-4858f64.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18010/HighFidelity-Beta-PR15141-4858f64.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18010/HighFidelity-Quest-Beta-PR15141-4858f64.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18010/HighFidelity-Beta-PR15141-4858f64.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18010/HighFidelity-Beta-Interface-PR15141-4858f64.exe

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18019/HighFidelity-Beta-PR15141-df32b61.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18019/HighFidelity-Quest-Beta-PR15141-df32b61.apk)

File: interface/src/avatar/MyAvatar.cpp (hunk 1 above)
Comment (luiscuenca) [reply interface/src/avatar/MyAvatar.cpp]: Thanks Sam. It has been addressed.

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18020/HighFidelity-Beta-PR15141-8015056.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18020/HighFidelity-Quest-Beta-PR15141-8015056.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18020/HighFidelity-Beta-PR15141-8015056.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18020/HighFidelity-Beta-Interface-PR15141-8015056.dmg

**Comment by luiscuenca** [comment]
### TEST PLAN
- Load an avatar with flow bones. i.e. [this one](https://hifi-content.s3.amazonaws.com/luis/models/Jene_Flow_FromFST.fst)
- Load this version of flow app [flowAppCpp](https://hifi-content.s3.amazonaws.com/luis/flowFiles/flowAppCpp.js). It uses the new flow C++ version. Make sure it works correctly.

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18020/HighFidelity-Beta-PR15141-8015056.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18020/HighFidelity-Beta-Interface-PR15141-8015056.exe

**Pull request merged**